js/src/logger.ts

```diff
@@ -2970,6 +2970,11 @@ export function initLogger<IsAsyncFlush extends boolean = true>(
     project_name: projectName,
     project_id: projectId,
   };
+
+  // When initLogger is called, we actually enable the queue size limit. We do this because
+  // initLogger is used for production obsservability and we don't want to OOM customer's apps.
+  setQueueSizeLimitEnabled(true);
```

Comment:
it's kind of a bummer that we have to do this globally, rather than per object. I also don't think that this will work in cases where we use the SDK in a multi-tenant fashion (eg our SDK). In those cases, we create an independent `state` variable for each session, and each one has its own background logger. So maybe you could place this state on the background logger itself, and then run this on whatever the object's current `state` variable is?

Reply:
yes agreed. that's a better design.